middle_end/flambda2/to_cmm/to_cmm_effects.ml

```diff
@@ -19,20 +19,26 @@ type effects_and_coeffects_classification =
   | Pure
   | Effect
   | Coeffect_only
+  | Generative_duplicable
```

Comment:
This is a bit annoying because of the overloading of the "duplicable" term. Here it means that it's ok to duplicate the primitive because it has only generative (immutable) effects. But in other parts it refers to whether the primitive has placement `Strict` or `Delay`. For clarity, renaming the constructor to `Generative_immutable` might help.

Reply:
I've done the renaming.